### PR TITLE
ddl: fix issue of rename table with foreign key constraint

### LINKED ISSUE #38080: rename table should update foreign key constraint
## Dev Task

Rename table should update foreign key constraint, which is update child table foreign key info.

This is a sub task of https://github.com/pingcap/tidb/issues/36982

### PR DESCRIPTION
**Pull request by crazycs520**
### What problem does this PR solve?

Issue Number: close #38080

Problem Summary: Rename table should update foreign key constraint, which is update child table foreign key info.

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: ddl/table.go [hunk 1]
```diff
@@ -1075,10 +1083,57 @@ func checkAndRenameTables(t *meta.Meta, job *model.Job, oldSchemaID, newSchemaID
...
+		if childFKInfo == nil {
+			continue
+		}
+		childFKInfo.RefSchema = newDB.Name
+		childFKInfo.RefTable = newTableName
+		infos[childTableInfo.tblInfo.ID] = childTableInfo
+	}
+	infoList := make([]schemaIDAndTableInfo, 0, len(fkh.loaded))
+	for _, info := range fkh.loaded {
+		if info.tblInfo.ID == tblInfo.ID {
```
Comment (lcwangchao) [inline ddl/table.go:1131]: Will self table's reference be lost here?

Comment (crazycs520) [reply ddl/table.go:1131]: No, self table has already been updated, no need to update again.  the logic here is same with `adjustForeignKeyChildTableInfoAfterModifyColumn`

Comment (crazycs520) [reply ddl/table.go:1131]: image

**Review by lcwangchao — APPROVED** [review]

**Review by tangenta — APPROVED** [review]

**Review by hawkingrei — APPROVED** [review]

**Pull request merged**